HELM mode does not support parameters without value

This issue is related to **HELM** support as discussed here : https://github.com/fabric8io/fabric8-maven-plugin/issues/1628.

I try to run the ResourceMojo in version 4.2 with file template.yml as this:

```
kind: Template
parameters:
- name: PROJECT_NAMESPACE
- name: limits.memory
  value: "1400Mi"
- name: requests.memory
  value: "700Mi"
```

Please, notice that **the parameter PROJECT_NAMESPACE does not provide a value**.
When running, this stack is displayed:
```
Caused by: java.lang.NullPointerException
    at java.lang.String.replace (String.java:2240)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.resolveTemplateVariables (ResourceMojo.java:699)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.resolveTemplateVariablesIfAny (ResourceMojo.java:677)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.writeResources (ResourceMojo.java:662)
    at io.fabric8.maven.plugin.mojo.build.ResourceMojo.executeInternal (ResourceMojo.java:397)
    at io.fabric8.maven.plugin.mojo.AbstractFabric8Mojo.execute (AbstractFabric8Mojo.java:65)
```

This is not very clear to understand.

See https://github.com/fabric8io/fabric8-maven-plugin/blob/v4.2.0/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java#L699

This is due to the call to `text = text.replace(from, to);` with `to=null`

I suggest two approaches:

### OPTION 1 : throw an explicit exception
HOW : add just before `text = text.replace(from, to);` this code

```
if (to==null) {
  throw new MojoExecutionException("missing value for HELM template parameter " + parameter.getName()+" in file "+kubernetesYaml);
}
text = text.replace(from, to);
```

### OPTION 2 : consider missing value as empty string
It is hard for me to consider if it can be a real use case. I ask the community for it.

It could be considered risky to authorize it...

HOW : add just before `text = text.replace(from, to);` this code
```
if (to==null) {
  to="";
}
```

Thanks,

Ben.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@ After this we will switch probably to real [Semantic Versioning 2.0.0](http://se
 * Fixed image config in thorntail sample
 * Fix #1697: NullpointerException when trying to apply custom resources
 * Fix #1696: fmp not setting imagestreams resourceVersion properly.
+* Fix #1689: HELM mode does not support parameters without value
 * Fix #1676: Support for latest kubernetes client
 * Feature #1536: Java Image Builder Support
 * Fix #1704: fabric8-build failing on openshift

--- a/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java
+++ b/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java
@@ -696,6 +696,9 @@ public class ResourceMojo extends AbstractFabric8Mojo {
         for (io.fabric8.openshift.api.model.Parameter parameter : parameters) {
             String from = "${" + parameter.getName() + "}";
             String to = parameter.getValue();
+            if (to == null) {
+                throw new MojoExecutionException("Missing value for HELM template parameter " + from + " in " + kubernetesYaml);
+            }
             text = text.replace(from, to);
         }
         if (!original.equals(text)) {

